pkg/net/libp2p/authenticated_connection_test.go

```diff
@@ -311,6 +312,6 @@ func (mf *mockFirewall) updatePeer(
 	remotePeerNetworkPublicKey *libp2pcrypto.Secp256k1PublicKey,
 	meetsCriteria bool,
 ) {
-	x := remotePeerNetworkPublicKey.X.Uint64()
-	mf.meetsCriteria[x] = meetsCriteria
+	ecdsaPublicKey := (*secp.PublicKey)(remotePeerNetworkPublicKey).ToECDSA()
+	mf.meetsCriteria[ecdsaPublicKey.X.Uint64()] = meetsCriteria
```

Comment:
This should go through a key converter. We can convert to operator key and extract X.

Reply:
It's a general rule of thumb that errors shouldn't be ignored. I think it makes sense to return an error from `updatePeer` and `t.Fatal` in the caller code in case the error is not `nil`.

Reply:
Added error handling in 6580f281affee59e3f1154531324c54cf314ea7d.